### ISSUE TITLE
[new-arch] Progressbar misbehaves when too long

### ISSUE THREAD

**Issue by felipesanches** [googlefonts/fontbakery#1424]
@graphicore If you run fontbakery on a family with enough fonts that can cause the number of checks to exceed a single-line progressbar, then the whole output is messed up by the gradual redrawing of the progressbar.

Please try to see the output of:

```
fontbakery check-googlefonts ~/gfonts/ofl/firasans/*.ttf
```

It is ugly during execution and the end-result is unreadable because the actual log result messages are chopped in the terminal buffer by snippets of the partial results summary and progressbar content.

**Comment by graphicore** [comment]
I'll look into this soon. Thanks for reporting.

**Comment by felipesanches** [comment]
This is better, but I still see some glitches from time to time. But the output does not get corrupted and the glitches I see are really minimal, so they are not a major concern anymore. I think I'm ok to close this issue now. Feel free to reopen the issue if you think the minor aesthetical problem is unbearable.

**Comment by graphicore** [comment]
The glitches are because the TTY is not updating fast enough. I added some controls to throttle the output one time based, one input count based. But, it's not straight forward to print asynchronously in python, so we're always waiting for something that wants to print to then decide whether to flush the logs or not. Thus, to much can't be done with the throttles, because it often means we're waiting for a long running test, while we already collected some logs from tests before.
